src/main/java/de/dennisguse/opentracks/io/file/importer/KmzTrackImporter.java

```diff
@@ -194,7 +194,7 @@ private boolean hasImageExtension(String fileName) {
      * @param trackId the id of the Track.
      */
     private void deleteOrphanImages(Context context, Track.Id trackId) {
-        if (!trackId.isValid()) {
+        if (trackId != null && !trackId.isValid()) {
```

Comment:
This should not be necessary as `findAndParseKmlFile()` either throws an exception or returns a trackid, right?

Reply:
I came across this when I first experimented with a kmz file for the Unit test. But I had problems with it an the while ZipEntry didn't find any files in the zip input stream. Then it resulted in a null trackId.

Reply:
Then we should throw an exception in `findAndParseKmlFile` as the file is broken or am I mistaken?

Reply:
Yes, you are right.